### PR TITLE
Fix max ballista shots

### PR DESCRIPTION
**Pull request by tau3**
Hi! Here is a fix for #4024

### CODE REVIEW

File: src/fheroes2/battle/battle_catapult.cpp [hunk 1]
```diff
@@ -56,6 +56,11 @@ Battle::Catapult::Catapult( const HeroBase & hero )
 
     if ( hero.HasArtifact( Artifact::BALLISTA ) )
         catShots += Artifact( Artifact::BALLISTA ).ExtraValue();
+
+    const int maxCatShots = 3;
+    if ( catShots > maxCatShots ) {
+        catShots = maxCatShots;
+    }
```
Comment (qarkai) [inline src/fheroes2/battle/battle_catapult.cpp:63]:
```suggestion
    catShots = std::max( catShots, maxCatShots );
```

Comment (oleg-derevenetz) [inline src/fheroes2/battle/battle_catapult.cpp:63]: Hi @tau3 may be just change `extra` value for `Ballista of Quickness` in `resource/artifact.cpp` from 2 to 1? Hi @Branikolog I suppose this artifact should just give one extra shot?

Comment (tau3) [reply src/fheroes2/battle/battle_catapult.cpp:63]: Hi! I was afraid of changing that static configuration and went the other way, but yes, your suggestion is better :)
Lets wait for a response

Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_catapult.cpp:63]: @tau3 This artifact has a misleading description - what does "lets your catapult fire twice per round" mean: is this mean "one additional shot" or "two additional shots".

**Comment by Branikolog** [comment]
Hi, @oleg-derevenetz !
In the original game this artifact adds I've extra shot, allowing to shot twice. If hero has expert Ballistics, catapult shots 3 times.

**Comment by oleg-derevenetz** [comment]
Hi @Branikolog

> In the original game this artifact adds I've extra shot, allowing to shot twice. If hero has expert Ballistics, catapult shots 3 times. 

So it seems that currently this artifact has wrong `extra` value of 2, and we can replace it by 1 instead...

**Comment by Branikolog** [comment]
> Hi @Branikolog
> 
> > In the original game this artifact adds I've extra shot, allowing to shot twice. If hero has expert Ballistics, catapult shots 3 times.
> 
> So it seems that currently this artifact has wrong `extra` value of 2, and we can replace it by 1 instead...

Correct!
Even without "Ballistics" catapult shoots 3 times.

**Comment by oleg-derevenetz** [comment]
@Branikolog 

> Correct!
> Even without "Ballistics" catapult shoots 3 times.

So with this artifact catapult always shoots 3 times, and Ballistics skill doesn't matter?

**Comment by Branikolog** [comment]
> @Branikolog
> 
> > Correct!
> > Even without "Ballistics" catapult shoots 3 times.
> 
> So with this artifact catapult always shoots 3 times, and Ballistics skill doesn't matter?

Yes. In the OG Catapult shoots 3 times maximum. 
4 shots looks kind of unfair. )

**Comment by LeHerosInconnu** [comment]
Hello everyone,

> @tau3 This artifact has a misleading description - what does "lets your catapult fire twice per round" mean: is this mean "one additional shot" or "two additional shots".

The "Ballista of Quickness" gives "one additional shot".

Here's how it should work:

Ballistics; Increases the catapult's shots, damage and accuracy in siege combat.
Basic Level: 1 shot with extra damage.
Advanced Level: 2 shots with extra damage.
Expert Level: 2 shots with max damage.

Without Ballistics: 1 shot.
Without Ballistics and with Ballista of Quickness: 1 shot + 1 shot = 2 shots.
With Ballistics Basic: 1 shot with extra damage.
With Ballistics Basic and Ballista of Quickness: 1 shot with extra damage + 1 shot = 2 shots with extra damage.
With Balllistics Advanced: 2 shots with extra damage
With Ballistics Advanced and Ballista of Quickness: 2 shots with extra damage + 1 shot = 3 shots with extra damage.
With Ballistics Expert: 2 shots with max damage.
With Ballistics Expert and Ballista of Quickness: 2 shots with max damage + 1 shot = 3 shots with max damage.

The extra value for "Ballista of Quickness" should be set to "1".
When the hero has Ballistics, logically, the Ballista of Quickness's extra shot should also have the extra damage bonus and the maximum damage bonus.

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by ihhub** [comment]
@tau3 , thank you very much for the fix!

**Pull request merged**